### PR TITLE
created API endpoint to calculate BMI

### PR DESCRIPTION
**Pull request by carolina-museum**
created API endpoint to calculate BMI, as a practice for Carolina to understand and get used to the code.

### CODE REVIEW

File: app/api/main.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 from fastapi import APIRouter
 
-from app.api.routes import chat
+from app.api.routes import chat
```
Comment (huyfififi) [inline app/api/main.py:3]: `nit`
https://peps.python.org/pep-0008/#other-recommendations
> Avoid trailing whitespace anywhere. Because it’s usually invisible, it can be confusing: e.g. a backslash followed by a space and a newline does not count as a line continuation marker. Some editors don’t preserve it and many projects (like CPython itself) have pre-commit hooks that reject it.

File: app/api/routes/chat.py [hunk 2]
```diff
@@ -53,6 +55,16 @@ def query(chat_request: ChatRequest):
     )
     return ChatResponse(message=response.choices[0].message.content)
 
+@router.post("/bmi", response_model=BMIResponse, status_code=200)
+def bmi(bmi_request: BMIRequest):
```
Comment (gomyway1216) [inline app/api/routes/chat.py:59]: please add it to the new file as this is not related to chat feature right now.

**Review by gomyway1216 — CHANGES_REQUESTED** [review]

**Comment by huyfififi** [comment]
Sorry maybe there are some problems in PEP8 check in GitHub. I will go ahead and check if it is working.
In the meantime, please run `black` before committing your changes.

```
$ black models.py
$ black models.py --diff --color  # showing diffs without making changes
```

File: app/api/main.py (hunk 1 above)
Comment (carolina-museum) [reply app/api/main.py:3]: Thank you for your review! Yes, in fact, I forgot to run ruff and black on this code. 
I will read the code styling guideline, pay attention while coding, and check my codes using tools.

File: app/models.py [hunk 3]
```diff
@@ -11,3 +11,11 @@ class ChatResponse(BaseModel):
 
 class LearnRequest(BaseModel):
     text: str = Field(description="The text to be learned by the chatbot")
+
+#Practice
```
Comment (huyfififi) [inline app/models.py:15]: `nit` top-level classes or functions should be surrounded by 2 blank lines
https://peps.python.org/pep-0008/#blank-lines

I have no idea why `ruff` does not raise the formatting error, but you might want to install `flake8` and use it as a linter for now.

File: app/models.py [hunk 4]
```diff
@@ -11,3 +11,11 @@ class ChatResponse(BaseModel):
...
+class BMIRequest(BaseModel):
+    weight: float = Field(description="Weight value requested to calculate BMI.")
+    height: float = Field(description="Height value requested to calculate BMI.")   
+    
+class BMIResponse(BaseModel):
+    message: str = Field(description="BMI value to be responded by the chatbot")
```
Comment (huyfififi) [inline app/models.py]: You don't have to update this now, but in general, backend engineers try to design APIs that are as generic as possible so that different kinds of clients (browsers, other servers) use them.

In this case, I would just return a float number and ask clients, such as browsers, to be responsible for formatting on UI, appending messages, or showing in colors.

File: app/models.py [hunk 5]
```diff
@@ -11,3 +11,13 @@ class ChatResponse(BaseModel):
...
+
+
+# Practice
+class BMIRequest(BaseModel):
+    weight: float = Field(description="Weight value requested to calculate BMI.")
+    height: float = Field(description="Height value requested to calculate BMI.")
+
+
+class BMIResponse(BaseModel):
+    message: str = Field(description="BMI value to be responded by the chatbot")
```
Comment (huyfififi) [inline app/models.py:23]: You don't have to update this now, but in general, backend engineers try to design APIs that are as generic as possible so that different kinds of clients (browsers, other servers) use them.

In this case, I would just return a float number and ask clients, such as browsers, to be responsible for formatting on UI, appending messages, or showing in colors.

File: app/api/main.py (hunk 1 above)
Comment (huyfififi) [reply app/api/main.py:3]: I think being aware of all the rules in PEP8 is very hard, especially for beginners 😅 
I would recommend to use tools like `flake8`, `ruff` (not working? 🤔 ), or `black` instead of reading the documentation.

**Comment by huyfififi** [comment]
Thanks for running `black` on `models.py`, and overall, your changes look good to me 👍

**Comment by huyfififi** [comment]
BTW, I can see that you are using two different GitHub accounts. You might want to revoke the old SSH key, generate a new one, and register it on carolina-museum

**Comment by carolina-museum** [comment]
@huyfififi That is exactly what I am struggling with right now!! Thank you for catching it.
I was able to generate a new SSH key and register to carolina-museum, but I am struggling to find how to revoke the old SSH key. 
Do you know how to do this? I will keep trying otherwise.

**Comment by huyfififi** [comment]
@carolina-museum have you deleted the one in @coguchi ?

**Comment by carolina-museum** [comment]
@huyfififi No, I haven't. I am still figuring how I can do this..

**Comment by carolina-museum** [comment]
@huyfififi , @gomyway1216  helped me with this problem. I can now use carolina-museum only.
Thank you all!!!!

**Comment by huyfififi** [comment]
Screenshot 2024-10-17 at 11 30 36 AM

can you mark this pull request as "draft" while making the requested changes? once you are done, you can mark the pull request "ready for review" again and re-request reviews

**Comment by carolina-museum** [comment]
Thank you! I will do that!

**Review by huyfififi — APPROVED** [review]
👍

**Review by gomyway1216 — APPROVED** [review]
